src/utils/timezones.ts

```diff
@@ -0,0 +1,141 @@
+        offsetMinutes < 10 ? `0${offsetMinutes}` : offsetMinutes
+    }`;
+}
+
+/**
+ * Return the offset from UTC in minutes.
+ */
+function offset(date: Date, timeZone: string): number {
+    // use Japanese with toLocaleString because they have a nice arrangement of date parts from largest to smallest.
+    // If toLocaleString on this engine does not support locale or options parameters, then this call will throw
+    // and we will catch that
+    const arr: number[] = date
+        .toLocaleString('ja', { timeZone })
+        .split(/[/\s:]/)
+        .map(part => {
+            return Number(part);
+        });
+    arr[1] -= 1; // months are zero-based in JS for some reason, so subtract one to get the right month number
+    const t1 = Date.UTC.apply(null, arr as [number, number, number, number, number, number]);
+    const t2 = new Date(date).setMilliseconds(0);
+    return (t2 - t1) / 60000;
+}
+
+/**
+ * Calculate the display name of the zone for the given date. If the zone is in DST on that date or not,
+ * then return the string appropriately. If the zone does not use DST or if the platform cannot calculate it,
+ * then just return the static displayName string from @box/cldr-data
+ */
+function getCurrentDisplayName(date: Date, zone: TimeZone): string {
+    const { name, displayName } = zone;
+    if (supportsIntl && zone.abbreviationDaylight) {
+        try {
+            const jan1 = new Date(date.getFullYear(), 0, 1, 0, 0, 0);
+            const jun1 = new Date(date.getFullYear(), 5, 1, 0, 0, 0);
+            const jan1offset = offset(jan1, name);
+            const jun1offset = offset(jun1, name);
+            if (jan1offset === jun1offset) {
+                // no DST in this zone in this year, so just return the already calculated display name
+                return displayName;
+            }
+
+            const current = offset(new Date(Date.now()), name);
+            let zonename = `GMT${getOffsetStr(current)} ${zone.nameLocalized} `;
+            if (jan1offset < jun1offset) {
+                // southern hemisphere
+                zonename += current === jan1offset ? zone.abbreviationDaylight : zone.abbreviationStandard;
+            } else {
+                // northern hemisphere
+                zonename += current === jun1offset ? zone.abbreviationDaylight : zone.abbreviationStandard;
+            }
+            return zonename;
+        } catch (e) {
+            // sometimes the current platform does not support all of the zones or toLocaleString does not support
+            // locale and option parameters, so we have to catch for those cases
+            return displayName;
+        }
+    } else {
+        return displayName;
+    }
+}
+
+function timezones(options: TimeZoneOptions = {}): TimeZoneEntries {
+    const zones: TimeZoneEntries = {};
+
+    const { includeZones = 'enabled' }: { includeZones?: string } = options;
```

Comment:
we force it to be `all` in the call and yet here we're setting it to `enabled` as a default. is there any reason for this?

Reply:
If you mean that we use `all` in the `findTimeZone` function, that's correct. The `findTimeZone` function searches all zones for the one with the given id. The `timezones` function itself (which is the default export) only returns the Box-enabled time zones (ie. the ones that are used in the Box locales) by default. You can get the full list only by passing in `includeZones: "all"`